homeassistant/components/sonos/__init__.py

```diff
@@ -152,7 +152,10 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     _LOGGER.debug("Reached async_setup_entry, config=%s", config)
 
     if advertise_addr := config.get(CONF_ADVERTISE_ADDR):
-        soco_config.EVENT_ADVERTISE_IP = advertise_addr
+        parts = advertise_addr.split(":")
```

Comment:
How would this work for IPv6?

Reply:
My recommendation would be to put the port in a separate parameter "advertise_port" and avoid the parsing.